src/reducers/collections.js

```diff
@@ -51,8 +52,21 @@ function validateCollection(configCollection) {
     // Cannot set custom delimiter without explicit and proper frontmatter format declaration
     throw new Error(`Please set a proper frontmatter format for collection "${name}" to use a custom delimiter. Supported frontmatter formats are yaml-frontmatter, toml-frontmatter, and json-frontmatter.`);
   }
+  if ((!!folder) && (selectIdentifier(fields.map(f => f.name)) === undefined)) {
+    // Verify that folder-type collections have a "slug"-type field.
+    throw new Error(`Collection "${name}" must have a field that is a valid entry identifier. Supported fields are ${validIdentifierFields.join(',')}`);
+  }
 }
 
+const validIdentifierFields = ["title", "path"];
+export const selectIdentifier = (entryData) => {
```

Comment:
`fieldNames` would be more self documenting than `entryData`, which I would expect to be a map of the actual entry data.